content/en/blog/general/generate-vaps/index.md

```diff
@@ -0,0 +1,357 @@
+linkTitle: "Generating Kubernetes ValidatingAdmissionPolicies from Kyverno Policies"
+author: Mariam Fahmy
+description: "Generating Kubernetes ValidatingAdmissionPolicies from Kyverno Policies"
+---
+In the previous blog post, we discussed writing [Common Expression Language (CEL)](https://github.com/google/cel-spec) expressions in Kyverno policies for resource validation. CEL was first introduced to Kubernetes for the Validation rules for CustomResourceDefinitions, and then it was used by Kubernetes ValidatingAdmissionPolicies in 1.26.
+
+> ValidatingAdmissionPolicies offer a declarative, in-process alternative to validating admission webhooks.
+
+> ValidatingAdmissionPolicies use the Common Expression Language (CEL) to declare the validation rules of a policy. Validation admission policies are highly configurable, enabling policy authors to define policies that can be parameterized and scoped to resources as needed by cluster administrators.
+
+This post will show you how to generate Kubernetes ValidatingAdmissionPolicies and their bindings from Kyverno policies.
+
+## Prerequisite
+Generating Kubernetes ValidatingAdmissionPolicies require the following:
+1. A cluster with Kubernetes 1.26 or higher.
+2. Enable the `ValidatingAdmissionPolicy` [feature gate](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/).
+3. Enable the `admissionregistration.k8s.io/v1beta1` API for v1.28.
+   OR 
+   Enable the `admissionregistration.k8s.io/v1alpha1` API for v1.26 and v1.27.
+4. Set the `--generateValidatingAdmissionPolicy` flag in the Kyverno admission controller.
+5. Grant the admission controller service account the required permissions to generate ValidatingAdmissionPolicies and their bindings.
+
+In this post, we will use the beta version of Kubernetes 1.28.
+
+## Installation & Setup
+1. Create a local cluster
+
+```bash
+kind create cluster --image "kindest/node:v1.28.0" --config - <<EOF
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+featureGates:
+  ValidatingAdmissionPolicy: true
+runtimeConfig:
+  admissionregistration.k8s.io/v1beta1: true
+  admissionregistration.k8s.io/v1alpha1: true
+nodes:
+  - role: control-plane
+  - role: worker
+EOF
+```
+
+2. Add the Kyverno Helm repository.
+
+```bash
+helm repo add kyverno https://kyverno.github.io/kyverno/
+helm repo update
+```
+
+3. Create a new file that overrides the values in the chart.
+
+```bash
+cat << EOF > new-values.yaml
+features:
+  generateValidatingAdmissionPolicy:
+    enabled: true
+
+admissionController:
+  rbac:
+    clusterRole:
+      extraResources:
+      - apiGroups:
+          - admissionregistration.k8s.io
+        resources:
+          - validatingadmissionpolicies
+          - validatingadmissionpolicybindings
+        verbs:
+          - create
+          - update
+          - delete
+          - list
+EOF
+```
+
+4. Deploy Kyverno
+
+```bash
+helm install kyverno kyverno/kyverno -n kyverno --create-namespace --values new-values.yaml
```

Comment:
Just a helpful tip when writing things like this...

When you're recommending people install something, it's always a good idea to list a specific version as part of the command. This is so when future changes are made that make your instructions "break", by pinning to a specific version they'll always work.

Reply:
I fixed it. Thank you!